photutils/psf/models.py

```diff
@@ -4,17 +4,428 @@
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['FittableImageModel2D', 'NonNormalizable',
+           'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model', 'get_grouped_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model',
-           'get_grouped_psf_model']
+
+class NonNormalizable(Warning):
+    """
+    Used to undicate that a :py:class:`FittableImageModel2D` model is
+    non-normalizable.
+
+    """
+    pass
+
+
+class FittableImageModel2D(Fittable2DModel):
+    """
+    A fittable 2D model of an image allowing for image intensity scaling
+    and image translations.
+
+    This class takes 2D image data and computes the
+    values of the model at arbitrary locations (including at intra-pixel,
+    fractional positions) within this image using spline interpolation
+    provided by :py:class:`~scipy.interpolate.RectBivariateSpline`.
+
+    The fittable model provided by this class has three model parameters:
+    an image intensity scaling factor (`flux`) which is applied to
+    (normalized) image, and two positional parameters (`x_0` and `y_0`)
+    indicating the location of a feature in the coordinate grid on which
+    the model is to be evaluated.
+
+    If this class is initialized with `flux` (intensity scaling factor)
+    set to `None`, then `flux` is be estimated as ``|sum(data)|``.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array containing 2D image.
+
+    origin : tuple, None, optional
+        A reference point in the input image ``data`` array. When origin is
+        `None`, origin will be set at the middle of the image array.
+
+        If `origin` represents the location of a feature (e.g., the position
+        of an intensity peak) in the input ``data``, then model parameters
+        `x_0` and `y_0` show the location of this peak in an another target
+        image to which this model was fitted. Fundamentally, it is the
+        coordinate in the model's image data that should map to
+        coordinate (`x_0`, `y_0`) of the output coordinate system on which the
+        model is evaluated.
+
+        Alternatively, when `origin` is set to ``(0,0)``, then model parameters
+        `x_0` and `y_0` are shifts by which model's image should be translated
+        in order to match a target image.
+
+    normalize : bool, optional
+        Indicates whether or not the model should be build on normalized
+        input image data. If true, then the normalization constant (*N*) is
+        computed so that
+
+        .. math::
+            N \cdot C \cdot |\Sigma_{i,j}D_{i,j}| = 1,
+
+        where *N* is the normalization constant, *C* is correction factor
+        given by the parameter ``correction_factor``, and :math:`D_{i,j}` are
+        the elements of the input image ``data`` array.
+
+    correction_factor : float, optional
+        A strictly positive number that represents correction that needs to
```

Comment:
> Actually, if correction_factor is 2 then fitted flux will be 2x larger than the flux when correction_factor is 1.
> This sounds in line with your expectation: "I thought this was the correction factor in the sense that if it's bigger, the output value (at fixed flux) would be bigger."
> Last paragraph may be not very clear so I'll try to provide a new angle:

I like that new angle, that does help.  But above I was also saying it would be good to directly reference the above equation (in the `normalize` description), because this is all much clearer in equation-form. I'm just saying that my first guess at the convention for `C` was wrong because I *thought* it was the inverse of what it actually is, and that wouldn't have happened if I'd had some hint "look at the equation, because it's hard to say in words what this means"